pyrate/core/orbital.py

```diff
@@ -81,82 +81,93 @@ def remove_orbital_error(ifgs: List, params: dict) -> None:
+    if type(orbfitlksx) != int or type(orbfitlksy) != int:
+        msg = f"Multi-look factors for orbital correction should be of type: int"
+        raise OrbitalError(msg)
+    if degree not in [PLANAR, QUADRATIC, PART_CUBIC]:
+        msg = "Invalid degree of %s for orbital correction" % C.ORB_DEGREE_NAMES.get(degree)
+        raise OrbitalError(msg)
+    if method not in [NETWORK_METHOD, INDEPENDENT_METHOD]:
+        msg = "Invalid method of %s for orbital correction" % C.ORB_METHOD_NAMES.get(method)
+        raise OrbitalError(msg)
+
+    # Give informative log messages based on selected options
+    log.info(f'Calculating {__degrees_as_string(degree)} orbital correction using '
+             f'{__methods_as_string(method)} method')
+    if orbfitlksx > 1 or orbfitlksy > 1:
+        log.info(f'Multi-looking interferograms for orbital correction with '
+                 f'factors of X = {orbfitlksx} and Y = {orbfitlksy}')
 
     if method == INDEPENDENT_METHOD:
-        log.info('Calculating orbital correction using independent method')
-        #TODO: implement multi-looking for independent orbit method
-        if params[C.ORBITAL_FIT_LOOKS_X] > 1 or params[C.ORBITAL_FIT_LOOKS_Y] > 1:
-            log.warning('Multi-looking is not applied in independent orbit method')
-        ifgs = [shared.Ifg(p) for p in ifg_paths] if isinstance(ifgs[0], str) else ifgs
+        ifg0 = shared.Ifg(ifg_paths[0]) if isinstance(ifg_paths[0], str) else ifg_paths[0]
+        original_dm = get_design_matrix(ifg0, degree, offset)
 
         if params[C.PARALLEL]:
             Parallel(n_jobs=params[C.PROCESSES], verbose=50)(
-                delayed(independent_orbital_correction)(ifg, params) for ifg in ifgs
+                delayed(independent_orbital_correction)(ifg_path, original_dm, params) for ifg_path in ifg_paths
             )
         else:
-            process_ifgs = mpiops.array_split(ifgs)
-            for ifg in process_ifgs:
-                independent_orbital_correction(ifg, params=params)
+            process_ifg_paths = mpiops.array_split(ifg_paths)
+            for ifg_path in process_ifg_paths:
+                independent_orbital_correction(ifg_path, design_matrix=original_dm, params=params)
 
     elif method == NETWORK_METHOD:
-        log.info('Calculating orbital correction using network method')
         # Here we do all the multilooking in one process, but in memory
         # can use multiple processes if we write data to disc during
         # remove_orbital_error step
         # A performance comparison should be made for saving multilooked
         # files on disc vs in memory single process multilooking
         if mpiops.rank == MAIN_PROCESS:
-            mlooked = __create_multilooked_dataset_for_network_correction(params)
+            mlooked = __create_multilooked_datasets(params)
             _validate_mlooked(mlooked, ifg_paths)
             network_orbital_correction(ifg_paths, params, mlooked)
```

Comment:
multilooking the ifg datasets could be done in parallel (i.e. use array_split within `__create_multilooked_datasets`?)

Reply:
The gdal swig bindings prevent us from doing this. The multilooked ifgs are held in memory using in-memory tifs.

Reply:
I have added this explanation as a comment in the code